lib/session.js

```diff
@@ -26,18 +27,85 @@ function Session(host, auth, options) {
   this._callbacks = [];
 
   this._debug = debug('nodemiral:sess:' + host);
+
+  // null means it's not set
+  self._ssh_config = null;
+
+  this._getSshConfigInfo(this._options.config, function (err, config) {
+    if(!err) {
+      self._ssh_config = config;
+      return;
+    } else if(self._options.config) {
+      throw err;
+    } else {
+      self._ssh_config = {};
+    }
+  });
 }
 
+Session.prototype._getSshConfigInfo = function(file, callback) {
+  var file = file || '~/.ssh/config';
+  var host = this._host;
+
+  fs.readFile(file, {encoding: 'utf8'}, function(err, content) {
```

Comment:
Check wether we've started the process. If so, push all the callbacks into a array and call later when done.